### PR TITLE
cat: ensure serrno is always set

### PR DESCRIPTION
**Pull request by AtariDreams**
Set serrno to errno before the loop begins for the edge case that res0 is immediately NULL.

### CODE REVIEW

**Comment by ngie-eign** [comment]
Please change the email address to a non-users.noreply.github.com based address.

**Comment by AtariDreams** [comment]
> Please change the email address to a non-users.noreply.github.com based address.

Done!

File: bin/cat/cat.c [hunk 1]
```diff
@@ -447,11 +446,14 @@ udom_open(const char *path, int flags)
 		errno = EINVAL;
 		return (-1);
 	}
-	cap_rights_init(&rights, CAP_CONNECT, CAP_READ, CAP_WRITE,
-	    CAP_SHUTDOWN, CAP_FSTAT, CAP_FCNTL);
+	cap_rights_init(&rights, CAP_CONNECT, CAP_READ, CAP_WRITE, CAP_SHUTDOWN,
+	    CAP_FSTAT, CAP_FCNTL);
```
Comment (bsdimp) [inline bin/cat/cat.c:450]: This change is cosmetic. Please submit is separately.

File: bin/cat/cat.c [hunk 2]
```diff
@@ -447,11 +446,14 @@ udom_open(const char *path, int flags)
...
+
+	/* Default error if something goes wrong. */
+	serrno = EINVAL;
+
 	for (res = res0; res != NULL; res = res->ai_next) {
-		fd = socket(res->ai_family, res->ai_socktype,
-		    res->ai_protocol);
+		fd = socket(res->ai_family, res->ai_socktype, res->ai_protocol);
```
Comment (bsdimp) [inline bin/cat/cat.c:456]: ditto

**Comment by bsdimp** [comment]
OK. Please drop the style changes I've highlighted. the other style changes should be mentioned in the commit message. Like "Simplify structure of code because fd < 0 test is not needed if we return early." or some such.

**Comment by AtariDreams** [comment]
@bsdimp Done!

**Comment by bsdimp** [comment]
landed as 38f57faa06cf

**Pull request closed without merging**